index.js

```diff
@@ -0,0 +1,13 @@
+function calculateTotal(items) {
+  items.forEach(item => Calculate(item))
+
+  return items
+}
+
+function Calculate(item) {
+  item.total = parseFloat(((item.price - (item.price * item.discount)) * item.quantity).toFixed(2))
+
+  return item
```

Comment:
This function should be pure. Using item.total makes it not.

```JS
function calculateItemTotal(item) {
  const unformattedTotal = item.price * (1 - item.discount) * item.quantity
  const total = parseInt(unformattedTotal * 100) / 100

  return { ...item, total }
}
```

Reply:
I will look into it.